RavenDB-13602 HardResetToNewCluster remove the irrelevant nodes from …

karmeli87 opened the pull request:
…database groups

grisha-kotler commented:
Can we have a test for this?

ayende commented on src/Raven.Server/ServerWide/ClusterStateMachine.cs:
```diff
@@ -1419,59 +1419,74 @@ public override void Initialize(RachisConsensus parent, TransactionOperationCont
         private void OnStateChange(object sender, RachisConsensus.StateTransition transition)
         {
-            if (transition.From == RachisState.Passive && transition.To == RachisState.LeaderElect)
+            if (transition.To != RachisState.LeaderElect)
```
This happens as soon as we move to `Passive` state.
But previously, that was when we moved from `Passive` to something else.

How do you handle removal of node from cluster and then adding it again in this case?

karmeli87 committed:
RavenDB-13602 when HardResetToNewCluster called, other nodes will remove that node from their topology

karmeli87 committed:
RavenDB-13602 HardResetToNewCluster remove the irrelevant nodes from database groups

- handle license changes when switching to single leader
- validate the new tag and set it only if the tx is committed
